### PR TITLE
Refinements

### PR DESCRIPTION
**Pull request by bebop23**
-  [improve setup instructions and create-wallet signature rejection](https://github.com/Michi-App/michi/commit/d8fd09c8537ddd787ff0ad799a5eb5d3977dcf12)
- [refresh data correctly when deposits or withdrawals are made](https://github.com/Michi-App/michi/commit/f2a43c707b2d554de72e0ef65dc38c093cc042a6)
- [correctly detect when a new chest is created and refresh the data](https://github.com/Michi-App/michi/commit/701dc6fc3caf5d84c5f1f177b3482284c52bd66d)
- [ui: color and spacing adjustments](https://github.com/Michi-App/michi/pull/2/commits/0c6669c811b06035734cfbf77205d1f6621158e3)
image

- [feat: update to new Michi contract addresses](https://github.com/Michi-App/michi/pull/2/commits/0c4a519abd31403633000f53568456cdcf4abcd1)

### CODE REVIEW

File: frontend/src/features/WalletView.tsx [hunk 1]
```diff
@@ -94,43 +104,32 @@ export default function WalletView(
...
-    chainId: defaultChain.id,
-    address: selectedToken?.token_address,
-    abi: selectedToken && tokenABIs[selectedToken?.token_address],
-    eventName: "Transfer",
-    onLogs(logs) {
-      const transferResponse = (logs[0] as unknown as TransferLog).args;
-      const isWithdrawal = transferResponse.to === account.address
+
+  useEffect(() => {
+    if (isWithdrawalConfirming) {
```
Comment (glebshkut) [inline frontend/src/features/WalletView.tsx]: shouldn't `isWithdrawalConfirmed` be here?

File: frontend/src/features/CreateNewWallet.tsx [hunk 2]
```diff
@@ -2,43 +2,46 @@ import { abi, michiChestHelperAddress } from "@/constants/contracts/MichiChest";
...
-  const { writeContractAsync } = useWriteContract()
+  const { writeContractAsync, isPending, data: hash } = useWriteContract()
+  const { isLoading: isConfirming, isSuccess: isConfirmed, data } =
+  useWaitForTransactionReceipt({
+    hash,
+  })
   const account = useAccount();
+  const isLoading = isPending || isConfirming;
 
   const [isButtonLoading, setIsButtonLoading] = useState(false);
```
Comment (glebshkut) [inline frontend/src/features/CreateNewWallet.tsx]: it's unused, can be deleted

**Review by glebshkut — CHANGES_REQUESTED** [review]
Really good stuff, I love it. But noticed a few small things

File: frontend/src/features/WalletView.tsx [hunk 3]
```diff
@@ -249,11 +249,11 @@ export default function WalletView(
             }
           }}
         >
-          {isProcessing &&
+          {isPending || isConfirming &&
```
Comment (glebshkut) [inline frontend/src/features/WalletView.tsx]: Consider adding `isProcessing` here, cause there's no loading spinner appearing while withdrawing

File: frontend/src/features/WalletView.tsx (hunk 1 above)
Comment (bebop23) [reply frontend/src/features/WalletView.tsx]: yea, good catch

**Review by glebshkut — COMMENTED** [review]
Withdrawal still doesn't have a loading spinner, other than that LGTM

File: frontend/src/features/WalletView.tsx (hunk 3 above)
Comment (bebop23) [reply frontend/src/features/WalletView.tsx]: added

**Review by glebshkut — APPROVED** [review]

**Pull request merged**